src/presign/participant.rs

```diff
@@ -1082,11 +1086,13 @@ impl PresignKeyShareAndInfo {
         let order = k256_order();
         let g = CurvePoint::GENERATOR;
 
-        let mut delta: BigNumber = sender_r1_priv.gamma.modmul(&sender_r1_priv.k, &order);
+        let mut delta: BigNumber = sender_r1_priv
+            .gamma
+            .modmul(&sender_r1_priv.k.clone().into(), &order);
```

Comment:
Looks like you are turning `SecretBigNumber` -> `BigNumber` via a `clone` + `into` here. This will "leak" the `SecretBigNumber`.

Reply:
You want to pass in the underlying BigNumber via a reference instead.